src/internal/components/drag-handle/index.tsx

```diff
@@ -1,44 +1,83 @@
 // Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
-import React from 'react';
+import React, { forwardRef, useRef } from 'react';
 import clsx from 'clsx';
 
+import { IconProps } from '../../../icon/interfaces';
 import InternalIcon from '../../../icon/internal';
+import useForwardFocus from '../../hooks/forward-focus';
+import { useMergeRefs } from '../../hooks/use-merge-refs';
+import { DragHandleProps } from './interfaces';
+import { ResizeIcon } from './resize-icon';
 
 import styles from './styles.css.js';
 
-export interface DragHandleProps {
-  ariaLabel: string;
-  ariaDescribedby?: string;
-  disabled?: boolean;
-  onPointerDown?: React.PointerEventHandler;
-  onKeyDown?: React.KeyboardEventHandler;
-}
-
-export default function DragHandle({
-  ariaLabel,
-  ariaDescribedby,
-  disabled,
-  onPointerDown,
-  onKeyDown,
-}: DragHandleProps) {
-  return (
-    // We need to use a div with button role instead of a button
-    // so that Safari will focus on it when clicking it.
-    // (See https://bugs.webkit.org/show_bug.cgi?id=22261)
-    // Otherwise, we can't reliably catch keyboard events coming from the handle
-    // when it is being dragged.
-    <div
-      role="button"
-      tabIndex={0}
-      className={clsx(styles.handle, disabled && styles['handle-disabled'])}
-      aria-label={ariaLabel}
-      aria-describedby={ariaDescribedby}
-      aria-disabled={disabled}
-      onPointerDown={onPointerDown}
-      onKeyDown={onKeyDown}
-    >
-      <InternalIcon variant={disabled ? 'disabled' : undefined} name="drag-indicator" />
-    </div>
-  );
-}
+export { DragHandleProps };
+
+const DragHandle = forwardRef(
+  (
+    {
+      variant = 'drag-indicator',
+      size = 'normal',
+      ariaLabel,
+      ariaDescribedby,
+      ariaValue,
+      disabled,
+      onPointerDown,
+      onKeyDown,
+      className,
+    }: DragHandleProps,
+    ref: React.Ref<DragHandleProps.Ref>
+  ) => {
+    const dragHandleRefObject = useRef<HTMLElement>(null);
+    const dragHandleRef = useMergeRefs(dragHandleRefObject);
```

Comment:
what do we merging here?

Reply:
Good catch! There was the dragHandleRef originally but I had to replace it with the combination of the ref object and ref to use the useForwardFocus. While it is a common strategy in most components, the drag handle is still internal and does not feature the __internalRootRef - I will update it to only keep the ref object for now.